### DISCUSSION TITLE
Multiple 1-n relations on same model

### DISCUSSION THREAD

**Discussion by perroudsky** [prisma/prisma#13790] [Q&A] [answered]
Hi guys, 
I'm trying to make a model so that a user can own multiple _Titles_ and equip one of them.

The following is how I've built it :

```prisma

model User {
  id          String   @id @default(cuid())
  titles            Title[]    @relation("titles")
  currentTitle      Title?     @relation("currentTitle", fields: [currentTitleId], references: [id])
  currentTitleId    String?
}

model Title {
  id           String  @id @default(cuid())
  name         String?
  users        User[]  @relation("titles")
  currentUsers User[]  @relation("currentTitle")
}

```

The problem is that it does register on the title side, but it does not on the individual user side when I try to query its current title.

Is it even possible ? Any help greatly appreciated :)

**Comment by austincrim** [comment, marked as answer]
Hey @perroudsky 👋,

Could you give me an example of how you are trying to query this model? That will help me understand what exactly you are trying to achieve 🙂.

**Reply by perroudsky** [reply]
Hi Austin !

So just to make my original post clearer :
I'm trying to make a model so that a user can own multiple Titles and _have one of them set as currentTitle_

Of course, it's just a basic query like so :
```gql
query {
  user(id: "ID1"){
    id
    titles {
      id
      name
    }
    currentTitle {
      id
      name
    }
  }
}
```
the one above return null as currentTitle

but if I fetch the titles, it does show the connection between the title and the user:
```gql
query {
  titles {
    id
    name
    currentUsers {
      id #ID1
      name
    }
  }
}
```

This is the User type :
```js
 import { objectType } from 'nexus'

export const User = objectType({
  name: 'User',
  definition(t) {
    t.string('id'),
    t.field('currentTitle', {
      type: 'Title',
      resolve: async (parent, args, ctx) => {
        const res = await ctx.prisma.user
          .findUnique({
            where: {
              id: parent.id || undefined,
            },
          })
          .currentTitle()

          // if I comment out the ".currentTitle()" line to only print the user,
          // currentTitleId is null
        return res
      },
    }),
    t.list.field('titles', {
      type: 'Title',
      resolve: (parent, args, ctx) => {
        return ctx.prisma.user
          .findUnique({
            where: {
              id: parent.id || undefined,
            },
          })
          .titles()
      },
    })
  },
})
```

and this is the way I create the relation :
```js
 t.field('setUserCurrentTitle', {
      type: 'User',
      args: {
        userId: stringArg(),
        titleId: stringArg(),
      },
      resolve: async (_, args, { prisma }) => {

        try {
          const title = await prisma.user.update({
            data: {
              currentTitle: {
                connect: { id: args.titleId },
              },
            },
            where: { id: args.userId },
          })
          return title
        } catch (e) {
          console.log(e)
        }
      },
    })

```

Hope this is clearer and if not I'd be happy to explain more :)

**Reply by perroudsky** [reply]
Oops I made it work
My bad it was a very silly mistake from somewhere else !

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
